src/Microsoft.Diagnostics.Runtime/src/MacOS/Structs/MachHeader64.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Diagnostics.Runtime.MacOS
+{
+    [StructLayout(LayoutKind.Sequential, Pack = 1)]
+    internal readonly struct MachHeader64
+    {
+        public static unsafe uint Size => (uint)sizeof(MachHeader64);
+        public const uint Magic64 = 0xfeedfacf;
```

Comment:
MachO files could also be the `universal`/`fat` binary format, where the first bytes are defined the [fat_header](https://github.com/apple/darwin-xnu/blob/main/EXTERNAL_HEADERS/mach-o/fat.h#L48-L54), followed by per-CPU architecture headers.